controller/src/plan.rs

```diff
@@ -20,10 +24,28 @@ impl ControllerPlan {
         let nats = config.nats.connect_with_retry().await?;
 
         let scheduler_plan = config.scheduler.map(|_| SchedulerPlan);
-        let dns_plan = config.dns.map(|options| DnsPlan {
-            options,
-            nc: nats.clone(),
-        });
+        let dns_plan = if let Some(options) = config.dns {
+            let soa_email = if let Some(soa_email) = options.soa_email {
+                let soa_email = format_rname(&soa_email).context(
```

Comment:
might it be better to sanitize the configuration in a separate function? or pass what this does to serde somehow?

Reply:
`format_rname` does a basic check (ensures that there is an `@`, basically). Full email validity checking is [famously crazy](http://www.ex-parrot.com/~pdw/Mail-RFC822-Address.html) and not really necessary here.